public/app/core/components/Select/SortPicker.tsx

```diff
@@ -7,37 +7,46 @@ import { DEFAULT_SORT } from 'app/features/search/constants';
 
 import { SearchSrv } from '../../services/search_srv';
 
-const searchSrv = new SearchSrv();
-
 export interface Props {
   onChange: (sortValue: SelectableValue) => void;
   value?: string;
   placeholder?: string;
+  getSortOptions?: () => Promise<SelectableValue[]>;
   filter?: string[];
 }
 
-const getSortOptions = (filter?: string[]) => {
-  return searchSrv.getSortOptions().then(({ sortOptions }) => {
-    const filteredOptions = filter ? sortOptions.filter((o: any) => filter.includes(o.name)) : sortOptions;
-    return filteredOptions.map((opt: any) => ({ label: opt.displayName, value: opt.name }));
+const defaultSortOptionsGetter = (): Promise<SelectableValue[]> => {
+  return new SearchSrv().getSortOptions().then(({ sortOptions }) => {
+    return sortOptions.map((opt: any) => ({ label: opt.displayName, value: opt.name }));
   });
 };
 
-export const SortPicker: FC<Props> = ({ onChange, value, placeholder, filter }) => {
+export function SortPicker({ onChange, value, placeholder, filter, getSortOptions }: Props) {
```

Comment:
@axelavargas this is really just a refactor that lets us pass in `getSortOptions` -- it means we have to move the filter into the `useAsync` below.